src/js/trove/parse-pyret.js

```diff
@@ -970,20 +970,24 @@
         },
         'dim-expr': function(node) {
           // (LANGLE|LT) unit-expr (RANGLE|GT) | (LANGLE|LT) NUMBER (RANGLE|GT)
-          // TODO: Handle numbers here somehow
           return tr(node.kids[1])
         },
         'unit-atom': function(node) {
-          if (node.kids.length === 1) {
+          if (node.kids.length === 1 && node.kids[0].name === 'NAME') {
             return RUNTIME.getField(ast, 'u-base')
               .app(pos(node.pos), name(node.kids[0]))
+          } else if (node.kids.length === 1 && node.kids[0].name === 'NUMBER') {
+            if (node.kids[0].value !== '1') {
```

Comment:
This would probably be better handled as a well-formedness error later, rather than a parse-error now.  Also, `1%<3>` currently triggers an internal error rather than any clean error, so there's some more cleanup needed here.

Reply:
My only hold-up with making this a wf check is that we have to change `u-one(l)` into `u-num(l, n)`. Not a major issue, something just feels _weird_ about holding onto this value in the AST just for the sake of a wf check. Do you think it's worth holding onto?